File: backend/app/models/property.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    DateTime,
    Boolean,
    ForeignKey,
)

from sqlalchemy.orm import relationship

from datetime import datetime, UTC

from app.database.database import Base


class Property(Base):

    __tablename__ = "properties"

    id = Column(
        Integer,
        primary_key=True,
        index=True
    )

    title = Column(
        String,
        nullable=False
    )

    description = Column(
        String,
        nullable=True
    )

    price = Column(
        Integer,
        nullable=True
    )

    city = Column(
        String,
        nullable=True
    )

    rooms = Column(
        Integer,
        nullable=True
    )

    image_url = Column(
        String,
        nullable=True
    )

    status = Column(
        String,
        nullable=False,
        default="available",
        server_default="available",
)

    contact_name = Column(
       String,
       nullable=True
)

    phone = Column(
       String,
       nullable=True
)

    whatsapp = Column(
       String,
       nullable=True
)

    created_at = Column(
        DateTime,
        default=lambda: datetime.now(UTC)
    )

    images = relationship(
        "PropertyImage",
        back_populates="property",
        cascade="all, delete-orphan",
    )


class PropertyImage(Base):

    __tablename__ = "property_images"

    id = Column(
        Integer,
        primary_key=True,
        index=True
    )

    property_id = Column(
        Integer,
        ForeignKey("properties.id", ondelete="CASCADE"),
        nullable=False
    )

    url = Column(
        String,
        nullable=False
    )

    is_cover = Column(
        Boolean,
        default=False
    )

    sort_order = Column(
        Integer,
        default=0
    )

    created_at = Column(
        DateTime,
        default=lambda: datetime.now(UTC)
    )

    property = relationship(
        "Property",
        back_populates="images"
    )

File: frontend/types/property.ts
export type PropertyStatus =
  | "pending"
  | "available"
  | "reserved"
  | "rented"
  | "archived"

export type PropertyImage = {
  id: number
  url: string
  is_cover: boolean
  sort_order: number
}

export type Property = {
  id: number
  owner_id?: number | null
  title: string
  description: string | null
  price: number | null
  city: string | null
  rooms: number | null
  image_url: string | null
  status: PropertyStatus
  images?: PropertyImage[]
  contact_name?: string | null
  phone?: string | null
  whatsapp?: string | null
  avatar_url?: string | null
  last_verified_at?: string | null
  report_count?: number
}
